cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginProperties.java

```diff
@@ -33,19 +35,40 @@
 
   // Currently only support String->String map.
   private final Map<String, String> properties;
+  private final Set<String> macros;
+
+  private PluginProperties(Map<String, String> properties, @Nullable Set<String> macros) {
+    this.properties = properties;
+    this.macros = macros;
+  }
 
   public static Builder builder() {
     return new Builder();
   }
 
   private PluginProperties(Map<String, String> properties) {
-    this.properties = properties;
+    this(properties, null);
   }
 
   public Map<String, String> getProperties() {
     return properties;
   }
 
+  @Nullable
```

Comment:
its nice not having to deal with nulls. Can we check for null and return an empty set if so?